### PR TITLE
fix: plumb http agent into documentdb client

### PR DESCRIPTION
**Pull request by joshgummersall**
This is definitely the sketchiest implementation, though I've verified it with the locally installed dependencies. It's also very defensive and purely opt-in.

### CODE REVIEW

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts [hunk 1]
```diff
@@ -128,6 +133,15 @@ export class CosmosDbStorage implements Storage {
         }
 
         this.client = new DocumentClient(settings.serviceEndpoint, { masterKey: settings.authKey }, policy);
+
+        // Note: hack, however it works with our version
+        if (settings.agent) {
```
Comment (stevengum) [inline libraries/botbuilder-azure/src/cosmosDbStorage.ts:138]: Is there an issue tracking the fix for the agent and/or moving away from this hack in the future?

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts [hunk 2]
```diff
@@ -51,6 +52,10 @@ export interface CosmosDbStorageSettings {
      * @deprecated Please use [[CosmosDbPartitionedStorage]]. See https://github.com/microsoft/botframework-sdk/issues/5467
      */
     partitionKey?: string;
+    /**
+     * (Optional) http agent to use for outbound requests
```
Comment (stevengum) [inline libraries/botbuilder-azure/src/cosmosDbStorage.ts:56]: As to not back us into a corner with potentially weird backwards-compat/"which-write-wins" logic in the future, what's the "deprecation" story for this patch/hack?

**Review by stevengum — COMMENTED** [review]
Conceptually this sounds great, but I had some questions about the future. Is there a version of the Cosmos DB SDK for JS that we should bump to? Should we make breaking changes in our API now to catch-up and avoid potential technical debt?

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts (hunk 1 above)
Comment (joshgummersall) [reply libraries/botbuilder-azure/src/cosmosDbStorage.ts:138]: The `documentdb` package is [already deprecated](https://www.npmjs.com/package/documentdb), so we won't ever upgrade it. As I'm sure we have many legacy customers using this package, it seems reasonable to take this.

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts (hunk 2 above)
Comment (joshgummersall) [reply libraries/botbuilder-azure/src/cosmosDbStorage.ts:56]: See above re: deprecation.

**Review by stevengum — APPROVED** [review]
Comments addressed, my bad for not seeing which class was being updated.

:shipit:

**Pull request merged**